src/Features/LanguageServer/Protocol/Extensions/ProtocolConversions.cs

```diff
@@ -596,6 +597,18 @@ public static ProjectId ProjectContextToProjectId(ProjectContext projectContext)
                 debugName: projectContext.Id.Substring(delimiter + 1));
         }
 
+        public static async Task<DocumentOptionSet> FormattingOptionsToDocumentOptionsAsync(
+            LSP.FormattingOptions options,
+            Document document,
+            CancellationToken cancellationToken)
+        {
+            var documentOptions = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var updatedOptions = documentOptions
+                .WithChangedOption(Formatting.FormattingOptions.UseTabs, !options.InsertSpaces)
+                .WithChangedOption(Formatting.FormattingOptions.TabSize, options.TabSize);
```

Comment:
Does this also need to set Formatting.FormattingOptions.IndentationSize as well to match tab size? I wouldn't be surprised if VS Code doesn't support the distinction but usually you want these the same unless you have really good reasons not to do so.

Reply:
Ah I see. LSP doesn't support indent size currently, only tab size, but I'll update the code so the two match.